### ISSUE TITLE
Test build is failing

### ISSUE THREAD

**Issue by ElDeveloper** [biocore/scikit-bio#1713]
These are the two failures:

```
=================================== FAILURES ===================================
1020______________ [doctest] skbio.stats.distance._permdisp.permdisp _______________
1021130     ...                       ['s1', 's2', 's3', 's4', 's5', 's6'])
1022131     >>> grouping = ['G1', 'G1', 'G1', 'G2', 'G2', 'G2']
1023132 
1024133     Run PERMDISP using 99 permutations to caluculate the p-value:
1025134 
1026135     >>> from skbio.stats.distance import permdisp
1027136     >>> import numpy as np
1028137     >>> #make output deterministic, should not be included during normal use
1029138     >>> np.random.seed(0)
1030139     >>> permdisp(dm, grouping, permutations=99)
1031Differences (unified diff with -expected +actual):
1032    @@ -4,5 +4,5 @@
1033     number of groups                 2
1034     test statistic             1.03296
1035    -p-value                       0.35
1036    +p-value                       0.29
1037     number of permutations          99
1038     Name: PERMDISP results, dtype: object
1039
1040/home/travis/miniconda/envs/env_name/lib/python3.8/site-packages/skbio/stats/distance/_permdisp.py:139: DocTestFailure
1041________________ TestRDAResults_biplot_score.test_biplot_score _________________
1042
1043self = <skbio.stats.ordination.tests.test_redundancy_analysis.TestRDAResults_biplot_score testMethod=test_biplot_score>
1044
1045    def test_biplot_score(self):
1046    
1047        rda_ = rda(y=self.Y, x=self.X, scale_Y=False, scaling=1)
1048    
1049        # Load data as computed with vegan 2.4-3:
1050        # library(vegan)
1051        # data(varechem)
1052        # data(varespec)
1053        # rda_ = rda(X=varespec, Y=varechem, scale=FALSE)
1054        # write.table(summary(rda_, scaling=1)$biplot,
1055        #             'vare_rda_biplot_from_vegan.csv', sep=',')
1056        # write.table(summary(rda_, scaling=1)$sites,
1057        #                     'vare_rda_sites_from_vegan.csv', sep=',')
1058        # write.table(summary(rda_, scaling=1)$species,
1059        #                     'vare_rda_species_from_vegan.csv', sep=',')
1060        # write.table(summary(rda_, scaling=1)$constraints, #
1061        #                     'vare_rda_constraints_from_vegan.csv', sep=',')
1062        # write.table(summary(rda_, scaling=1)$cont$importance[2, ],
1063        #                     'vare_rda_propexpl_from_vegan.csv', sep=',')
1064        # write.table(summary(rda_, scaling=1)$cont$importance[1, ],
1065        #                     'vare_rda_eigvals_from_vegan.csv', sep=',')
1066    
1067        vegan_features = pd.read_csv(
1068            get_data_path('vare_rda_species_from_vegan.csv'))
1069        vegan_samples = pd.read_csv(
1070            get_data_path('vare_rda_sites_from_vegan.csv'))
1071        vegan_biplot = pd.read_csv(
1072            get_data_path('vare_rda_biplot_from_vegan.csv'))
1073        vegan_constraints = pd.read_csv(
1074            get_data_path('vare_rda_constraints_from_vegan.csv'))
1075        vegan_propexpl = pd.read_csv(
1076            get_data_path('vare_rda_propexpl_from_vegan.csv'))
1077        vegan_propexpl = pd.Series(
1078            vegan_propexpl.x.values, index=rda_.eigvals.index)
1079        vegan_eigvals = pd.read_csv(
1080            get_data_path('vare_rda_eigvals_from_vegan.csv'))
1081        vegan_eigvals = pd.Series(
1082            vegan_eigvals.x.values, index=rda_.eigvals.index)
1083    
1084        # scikit-bio returns singular values, whereas vegan returns eigenvalues
1085        vegan_eigvals = np.sqrt(vegan_eigvals*vegan_eigvals.shape[0])
1086        vegan_propexpl = vegan_eigvals/vegan_eigvals.sum()
1087    
1088        # transform the output of rda_ to match column selection of vegan
1089        res_samples = rda_.samples.iloc[:, 0:6]
1090        res_features = rda_.features.iloc[:, 0:6]
1091    
1092        rda_ = OrdinationResults(
1093            'RDA', 'Redundancy Analysis',
1094            samples=res_samples,
1095            features=res_features,
1096            sample_constraints=rda_.sample_constraints.iloc[:, 0:6],
1097            biplot_scores=rda_.biplot_scores.iloc[:, 0:6],
1098            proportion_explained=rda_.proportion_explained,
1099            eigvals=rda_.eigvals)
1100    
1101        exp = OrdinationResults(
1102            'RDA', 'Redundancy Analysis',
1103            samples=vegan_samples,
1104            features=vegan_features,
1105            sample_constraints=vegan_constraints,
1106            biplot_scores=vegan_biplot,
1107            proportion_explained=vegan_propexpl,
1108            eigvals=vegan_eigvals)
1109    
1110>       pdt.assert_frame_equal(res_samples, vegan_samples)
```

**Comment by thermokarst** [comment]
I seem to recall that @ebolyen & I ran into some issues with the permdisp doctest referenced above when we were cutting the last release.

**Comment by wasade** [comment]
I've started enumerating versions of pandas, numpy and scipy. Will hopefully have some insight here soon.

**Comment by wasade** [comment]
# vanilla install
numpy                     1.19.1           py36hbc911f0_0
numpy-base                1.19.1           py36hfa32c7d_0
scipy                     1.5.0            py36h0b6359f_0
pandas 1.1.0 (did not have tag on hand)

# pandas variation
1 failure with pandas 1.0.0
4 failures with pandas 1.1.0
1 failure with pandas < 1.0.0 (same as with 1.0.0)

# holding pandas at 1.0.0 (minimum supported version in skbio)
1 failure with scipy 1.3.0 (minimum supported version by skbio)
17 failures with numpy 1.14.6 (minimum supported by pandas 1.0.0)

**Comment by wasade** [comment]
I think [this](https://travis-ci.org/github/biocore/scikit-bio/jobs/653689236) is the last build that passed. I used the same version of numpy (1.18.1), pandas (1.0.1) and scipy (1.4.1) and still had a failure. Something weird is going on.

**Comment by wasade** [comment]
Just retested on Linux using a base environment derived from the last build that succeeded and still have not been able to get a pass.

```bash
$ conda create --name another-skbio-test python=3.6 glib=2.63.1 ipython=7.12.0 matplotlib=3.1.3 numpy=1.18.1 pandas=1.0.1 scikit-learn=0.22.1 scipy=1.4.1
$ source activate another-skbio-test
$ pip install -e .
$ pip install pytest
$ pytest
...snip...
============================================================== short test summary info ===============================================================
FAILED skbio/stats/ordination/tests/test_redundancy_analysis.py::TestRDAResults_biplot_score::test_biplot_score - AssertionError: DataFrame.iloc[:,...
======================================= 1 failed, 2321 passed, 30 skipped, 4602 warnings in 104.06s (0:01:44) ========================================
```

The dependency stack of scikit-bio is pretty large, and I'm a little uncertain about what else to attempt right now beyond a random walk.

@mortonjt, do you by chance have any insight into the `test_biplot_score` failure? The results appear to be `* -1` from expected. I tried `ignore_directionality=False` on the assertion as well, but it did not alter the result? Is the current result still correct, and if so should the test be updated to allow for a `* -1` result? An example of the failure can be seen [here](https://travis-ci.com/github/biocore/scikit-bio/jobs/340775586#L1147-L1153) from #1703

**Comment by mortonjt** [comment]
I'm not sure what happened upstream to cause this issue.  But I will say that these tests are very fragile and probably should be rewritten.  Minor upstream changes in the dependencies could explain the testing discrepancies.  Due to translation / rotation variance of these methods, the results may appear different even if the method is working as expected.

For now, fixing the sign issue should be ok. But it would be better if we could implement a pairwise sample-by-sample test to make sure that the distances are being calculated correctly.

**Comment by mortonjt** [comment]
Also want to mention that all of the tests are passing on my mac laptop.

**Comment by wasade** [comment]
Ya, it seems OS dependent. Do you have a quick fix that might be possible to put in here?

**Comment by mortonjt** [comment]
Sure thing.  See here: https://github.com/biocore/scikit-bio/pull/1720/
Let's see if the tests pass...

**Comment by rainarit** [comment]
Ya'll can check #1719 for additional the last make html pass (Sphinx<=3.0 is used)!

**Comment by ElDeveloper** [comment]
This is now addressed by #1719
